### PR TITLE
Refactor context

### PR DESCRIPTION
**Pull request by fabriziosestito**
Refactor to remove the clash between result/execution/execution result

It moves the execution server API inside the server module, so we can expose the context and refactor the schema with a proper name.

### CODE REVIEW

File: lib/wanda_web/controllers/result_controller.ex [hunk 1]
```diff
@@ -4,7 +4,7 @@ defmodule WandaWeb.ResultController do
 
   alias OpenApiSpex.Schema
```
Comment (arbulu89) [inline lib/wanda_web/controllers/result_controller.ex:5]: Should we still continue calling this `result_controller`? Everything was moved to `Executions` so maybe `executions_controller`. And together with this the view code, and most probably the routes (`/api/checks/executions`)

File: lib/wanda_web/views/result_view.ex [hunk 2]
```diff
@@ -16,11 +16,11 @@ defmodule WandaWeb.ResultView do
   end
 
   def render("result.json", %{
```
Comment (arbulu89) [inline lib/wanda_web/views/result_view.ex:18]: `execution.json` instead of `result.json`

**Review by arbulu89 — APPROVED** [review]
If the idea is to remove completely the `result` concept, we still have it in the code in some few places. I have noted some of them.

The worst thing is that it will open many conflicts with this hehe: https://github.com/trento-project/wanda/pull/63

File: lib/wanda/executions.ex [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
+    Target
+  }
 
   import Ecto.Query
 
   @doc """
   Create a new execution.
   """
-  @spec create_execution_result!(String.t(), String.t(), [Target.t()]) :: ExecutionResult.t()
+  @spec create_execution_result!(String.t(), String.t(), [Target.t()]) :: Execution.t()
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply lib/wanda/executions.ex:19]: grrr i missed it 
thanks

File: lib/wanda_web/controllers/result_controller.ex (hunk 1 above)
Comment (fabriziosestito) [reply lib/wanda_web/controllers/result_controller.ex:5]: @nelsonkopliku is taking care of this in another PR

**Review by nelsonkopliku — APPROVED** [review]
I saw this coming to life :smile: Looks good!

**Comment by fabriziosestito** [comment]
let's wait to merge this once we have the API pr merged

**Review by arbulu89 — APPROVED** [review]

**Pull request merged**